README.md

```diff
@@ -317,60 +316,59 @@ queryClient.setQueryData(
 
 ```
 
-### `createQueryOptions`
+### `createUseQueryOptions`
 
 ```ts
-function createQueryOptions<I extends Message<I>, O extends Message<O>>(
-  methodSig: MethodUnaryDescriptor<I, O>,
+function createUseQueryOptions<I extends DescMessage, O extends DescMessage>(
+  schema: MethodUnaryDescriptor<I, O>,
   input: SkipToken | PartialMessage<I> | undefined,
   {
     transport,
-    callOptions,
-  }: ConnectQueryOptions & {
+  }: {
     transport: Transport;
   },
 ): {
   queryKey: ConnectQueryKey<I>;
-  queryFn: QueryFunction<O, ConnectQueryKey<I>>;
-  enabled: boolean;
+  queryFn: QueryFunction<MessageShape<O>, ConnectQueryKey<I>> | SkipToken;
+  structuralSharing: Exclude<UseQueryOptions["structuralSharing"], undefined>;
 };
 ```
 
-A functional version of the options that can be passed to the `useQuery` hook from `@tanstack/react-query`. When called, it will return the appropriate `queryKey`, `queryFn`, and `enabled` flag. This is useful when interacting with `useQueries` API or queryClient methods (like [ensureQueryData](https://tanstack.com/query/latest/docs/reference/QueryClient#queryclientensurequerydata), etc).
+A functional version of the options that can be passed to the `useQuery` hook from `@tanstack/react-query`. When called, it will return the appropriate `queryKey`, `queryFn`, and `structuralSharing` flag. This is useful when interacting with `useQueries` API or queryClient methods (like [ensureQueryData](https://tanstack.com/query/latest/docs/reference/QueryClient#queryclientensurequerydata), etc).
 
 An example of how to use this function with `useQueries`:
 
 ```ts
 import { useQueries } from "@tanstack/react-query";
-import { createQueryOptions, useTransport } from "@connectrpc/connect-query";
+import { createUseQueryOptions, useTransport } from "@connectrpc/connect-query";
 import { example } from "your-generated-code/example-ExampleService_connectquery";
 
 const MyComponent = () => {
   const transport = useTransport();
   const [query1, query2] = useQueries([
-    createQueryOptions(example, { sentence: "First query" }, { transport }),
-    createQueryOptions(example, { sentence: "Second query" }, { transport }),
+    createUseQueryOptions(example, { sentence: "First query" }, { transport }),
+    createUseQueryOptions(example, { sentence: "Second query" }, { transport }),
   ]);
   ...
 };
 ```
 
-### `createInfiniteQueryOptions`
+### `createUseInfiniteQueryOptions`
```

Comment:
Oof, I just realized that this function is exported as `createInfiniteQueryOptions`. I'll revert.

What was the plan here, @paul-sachs?

Reply:
I was wondering why the function is exported with a different name. Why not rename the function if this name is a better fit?

Reply:
Ah, that I don't remember. No explicit reason. Renaming it internally definitely makes sense.

Reply:
👍 I'll do that separately.